File: Project1/project1.java
/**
 * 
 * 
 * @Jonathan Arenson
 * CS323-21
 * Project1
 * 
 */

import java.io.FileReader;
import java.io.IOException;
import java.io.*;
import java.io.PrintWriter;
import java.io.BufferedWriter;
import java.util.Scanner;

public class project1 {
	public static void main(String[] args){
		//this sets the input file as null
		Scanner input = null;
		//this sets the output file as null
		PrintWriter output = null;
		//initializes and declare all necessary variables

		//256 is the max number of ascii chars
		int[] charCounter = new int[256];
		String symbol;
		int index;
		char charIn;


		try{
			//allows user to enter the input text name
			input = new Scanner(new FileReader(args[0]));
			//allows users to enter to output text name
			output = new PrintWriter(args[1]);

		}

		/*catch(FileNotFoundException fnf){
			System.out.println(fnf);
		}*/

		catch(IOException io){
			System.out.println(io);
		}

		//This checks for all symbols
		input.useDelimiter("");

		while(input.hasNext()){
			//This continuously searches for more characters
			charIn = input.next().charAt(0);
			index = (int)charIn;

			if(index>32 && index<256){
				charCounter[index]++;
			}		
		}

		
		for(int i = 0; i<256; i++){

			if (charCounter[i]>0) {
				output.println((char)i+": "+charCounter[i]+" ");
			}
		}
		//closes input file
		input.close();
		//closes output file
		output.close();
	}	
}

File: Project1/project1.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

// Jonathan Arenson

void printAry(ofstream& outFile, int charCounter[]){
    int index = 0;
    char symbol;

    
   while(index < 256){

        if(charCounter[index]>0){
            symbol = (char)index;

                outFile<< symbol<< " " << ":" << charCounter[index] <<endl;
        }
    index++;

    }

}


int main(int argc, char* argv[]){
        ifstream inFile;
        ofstream outFile;
        //ofstream outFile;
        string str;
        int index;
        int charCounter[256] = {0};
        inFile.open (argv[1]);
        outFile.open (argv[2]);
        char charIn;
        if(inFile) {
          while(!inFile.eof()){
                inFile.get(charIn);
                
                index = (int)charIn;   
                charCounter[index]++;

    }
}


printAry(outFile, charCounter);


        inFile.close();
        outFile.close();
        return 0;

}


